packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,188 @@
+    availableUtxos: availableInputs,
+    requiredSigners,
+    totalSigners,
+    addressType: scriptType,
+  });
+
+  const analysis = txAnalyzer.analyze();
+
+  // Step 2: Verify CPFP suitability
+  if (!analysis.canCPFP) {
+    throw new Error(
+      "CPFP is not possible for this transaction. Ensure the specified output is available for spending.",
+    );
+  }
+
+  if (analysis.recommendedStrategy !== FeeBumpStrategy.CPFP) {
+    console.warn(
+      "CPFP is not the recommended strategy for this transaction. Consider using the recommended strategy: " +
+        analysis.recommendedStrategy,
+    );
+  }
+
+  // Step 3: Create a new transaction template for the child transaction
+  const childTxTemplate = new BtcTransactionTemplate({
+    inputs: [],
+    outputs: [],
+    targetFeeRate: Number(txAnalyzer.cpfpFeeRate), // Use CPFP-specific fee rate from analyzer
+    dustThreshold: Number(txAnalyzer.getDustThreshold),
+    network,
+    scriptType,
+    requiredSigners,
+    totalSigners,
+  });
+
+  // Step 4: Add the spendable output from the parent transaction as an input
+  const parentOutput = txAnalyzer.outputs[spendableOutputIndex];
+  if (!parentOutput) {
+    throw new Error(
+      `Spendable output at index ${spendableOutputIndex} not found in the parent transaction.`,
+    );
+  }
+
+  const parentOutputUTXO = availableInputs.find(
+    (utxo) => utxo.txid === analysis.txid && utxo.vout === spendableOutputIndex,
+  );
+
+  if (!parentOutputUTXO) {
+    throw new Error(
+      `UTXO for the spendable output (${analysis.txid}:${spendableOutputIndex}) not found in availableInputs.`,
+    );
+  }
+
+  childTxTemplate.addInput(
+    new BtcTxInputTemplate({
+      txid: analysis.txid,
+      vout: spendableOutputIndex,
+      amountSats: parentOutputUTXO.value,
+    }),
+  );
+
+  // Step 5: Add a change output (required for a valid transaction)
+  childTxTemplate.addOutput(
+    new BtcTxOutputTemplate({
+      address: changeAddress,
+      amountSats: 0, // Initial amount, will be adjusted later
+      type: TxOutputType.CHANGE,
+    }),
+  );
+
+  // Step 6: Add additional inputs if necessary to cover the fee
+  for (const utxo of availableInputs) {
+    if (childTxTemplate.feeRateSatisfied && childTxTemplate.areFeesPayPaid()) {
+      break; // Stop adding inputs once fee requirements are met
+    }
+    // Skip if this UTXO is already added (parent output) or is the parent transaction
+    if (utxo.txid === analysis.txid) {
+      continue;
+    }
+    childTxTemplate.addInput(
+      new BtcTxInputTemplate({
+        txid: utxo.txid,
+        vout: utxo.vout,
+        amountSats: utxo.value,
+      }),
+    );
+  }
+
+  // Step 7: Calculate and set the change output amount
+  const totalInputAmount = childTxTemplate.getTotalInputAmount();
+  const fee = childTxTemplate.targetFeesToPay;
+  const changeAmount = totalInputAmount - fee;
+
+  if (changeAmount <= Number(txAnalyzer.getDustThreshold)) {
+    throw new Error(
```

Comment:
I don't think this needs to be an error. You can just leave as dust, or donate to miners for extra fees with an amount of zero.

Reply:
I was thinking in some other contexts that a `strict` mode could be useful for things like this. If `strict` is on then it would throw here, if not then it can be flexible.

Reply:
added the new strict mode now :) [here](https://github.com/caravan-bitcoin/caravan/pull/114/commits/a215dea59e6ac918e960757d5dd6f74ed09ae55d)